apps/jetbrains/jetbrains.py

```diff
@@ -153,8 +154,18 @@ def idea_grab(times: int):
 
 @ctx.action_class("win")
 class win_actions:
+    def filename():
+        title = actions.win.title()
+        result = re.search(r"[A-Za-z-_]+\.java", title)
```

Comment:
I didn't catch this when initially poking at Adam's branch, but we'll want to take a moment to make this generic, and not java-specific. This does leave a lot of potential filenames out as-is - anything with a number etc.

I opted to avoid regex in e.g. vscode.
https://github.com/knausj85/knausj_talon/blob/cf723e1cbe934c65f8ed88fb32160b3e8de75bdd/apps/vscode/vscode.py#L38
since it always displays things like so

```
extension.ts - command-server - Visual Studio Code
```

Does jetbrains do something similar? I haven't used it in some time.

Reply:
you're right this needs to be generic... my excuse is I haven't coded in a long time :)

on windows and mac for java intellij:
`untitled - Main.java [optional-module-name]` or `untitled - Main.java`
 so it can either be the last or second to last.

on windows for python pycharm (haven't tried on mac but I assume its similar):
`pythonPython - main.py`

I will change so it checks either of the last 2 elements.

How come in your implementation for vscode you didn't split by space instead and just grab the first element. Then you won't need specified mac or not right?

Reply:
It simply didn't occur to me to do it that way for VSCode at the time. Derp!